pymeasure/instruments/andeenhagerling/ah2500a.py

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import math
+import re
+import logging
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range
+
+log = logging.getLogger(__name__)
+
+
+class AH2500A(Instrument):
+    """ Andeen Hagerling 2500A Precision Capacitance Bridge implementation
+    """
+    # regular expression to extract measurement values
+    _reclv = re.compile(r"[FHZ0-9.=\s]*C=\s*(-?[0-9.]+)\s*PF L=\s*(-?[0-9.]+)\s*NS V=\s*(-?[0-9.]+)\s*V")
+    _renumeric = re.compile(r'[-+]?(\d*\.?\d+)')
+
+    @classmethod
+    def _parse_reply(cls, string):
+        """
+        parse reply string from Andeen Hagerling capacitance bridges.
+
+        :param string: reply string from the instrument. This commonly is:
+          2500A:
+            "C= 1.234567    PF L= 0.000014    NS V= 0.750   V"
+          2700A:
+            "F= 1000.00 HZ C= 4.20188     PF L=-0.0260      NS V= 15.0     V"
+        """
+        m = cls._reclv.match(string)
+        if m is not None:
+            values = tuple(map(float, m.groups()))
+            return values
+        # if an invalid string is returned ('EXCESS NOISE')
+        if string.strip() == "EXCESS NOISE":
+            log.warning("Excess noise, check your experiment setup")
+            return (math.nan, math.nan, math.nan)
+        else:  # some unknown return string (e.g. misconfigured units)
+            raise Exception("Measurement value not understood")
+
+    config = Instrument.measurement(
+        "SHOW",
+        """ Read out configuration """,
+    )
+
+    caplossvolt = Instrument.measurement(
+        "Q",
+        """ Perform a single capacitance, loss measurement and return the
+        values in units of pF and nS. The used measurement voltage is returned
+        as third value.""",
+        get_process=lambda v: AH2500A._parse_reply(v),
```

Comment:
```suggestion
        get_process=AH2500A._parse_reply,
```
is a bit simpler. This is already a callable, no need to wrap in a lambda

Reply:
as you see in the CI runs below your suggested code fails. I think I also tested this before and it does not work because you can't use a class method here before the class definition is finished... Maybe there is a different way this can be written? I will revert back tot he lambda function in the meantime

Reply:
Yeah, sorry for not catching this -- sometimes coding "blind" and just making suggestions without trying them does not work. :-P
It's fine as it is. There are a couple of possibilities to do this differently, but they are not more elegant.